polyply/tests/test_presistence.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test that build files are properly read.
+"""
+import pytest
+import textwrap
+import math
+import numpy as np
+from vermouth.forcefield import ForceField
+import polyply
+from polyply.src.topology import Topology
+from polyply.src.top_parser import read_topology
+from polyply.src.nonbond_engine import NonBondEngine
+from polyply.src.build_file_parser import Presistence_specs
+from polyply.src.presistence import sample_end_to_end_distances
+
+
+@pytest.fixture
+def topology():
+    top_lines ="""
+    [ defaults ]
+    1   1   no   1.0     1.0
+    [ atomtypes ]
+    N0 72.0 0.000 A 0.0 0.0
+    [ nonbond_params ]
+    N0   N0   1 4.700000e-01    3.700000e+00
+    [ moleculetype ]
+    testA 1
+    [ atoms ]
```

Comment:
Should this inlcude at least one atom pair that is not N0-N0?

Reply:
Doesn't matter because the volumes are what determines the size of a particle in CG space and the volumes are set manually to different values for the different residues.